scalanet/src/io/iohk/scalanet/peergroup/dynamictls/CustomHandlers.scala

```diff
@@ -0,0 +1,33 @@
+package io.iohk.scalanet.peergroup.dynamictls
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.ChannelHandlerContext
+import io.netty.handler.ipfilter.AbstractRemoteAddressFilter
+
+import java.net.{InetAddress, InetSocketAddress}
+
+private[scalanet] object CustomHandlers {
+  // to share handlers between pipelines they need to be marked as @Sharable, if not netty refuses to share it.
+  @Sharable
+  class ThrottlingIpFilter(config: DynamicTLSPeerGroup.IncomingConnectionThrottlingConfig)
+      extends AbstractRemoteAddressFilter[InetSocketAddress] {
+    private val cache = Caffeine
+      .newBuilder()
+      .expireAfterWrite(config.throttlingDuration.length, config.throttlingDuration.unit)
+      .build[InetAddress, java.lang.Boolean]()
+    private val cacheView = cache.asMap()
```

Comment:
```suggestion
    private val cacheView = Caffeine
      .newBuilder()
      .expireAfterWrite(config.throttlingDuration.length, config.throttlingDuration.unit)
      .build[InetAddress, java.lang.Boolean]()
      .asMap()
```
Would it make any difference? At least it would be obvious that `cache` isn't used for anything.